CMakeLists.txt

```
@@ -537,12 +542,14 @@ if(CELERITAS_BUILD_DOCS)
     endif()
   endif()
 
-  set(_prev_mmdc "${MMDC_EXECUTABLE}")
-  find_program(MMDC_EXECUTABLE mmdc)
-  if(MMDC_EXECUTABLE AND NOT _prev_mmdc)
-    message(STATUS "Found mmdc: ${MMDC_EXECUTABLE}")
-  elseif(_prev_mmdc STREQUAL "")
-    message(STATUS "Could NOT find mermaid generator `mmdc`")
+  if(NOT DOXYGEN_DOT_EXECUTABLE)
+    find_program(DOXYGEN_DOT_EXECUTABLE dot)
+  endif()
+  if(DOXYGEN_DOT_EXECUTABLE AND NOT _prev_dot)
+    message(STATUS "Found dot: ${DOXYGEN_DOT_EXECUTABLE}")
+  elseif(_prev_dot STREQUAL "")
+    # Note: use string comparison, not boolean, to avoid matching old "NOTFOUND"
+    message(STATUS "Could NOT find graphviz generator `dot`")
   endif()
```

If I read this correctly: 
When `DOXYGEN_DOT_EXECUTABLE` is not found now but was found before (i.e. `_prev_dot` is set), there is no message **AND** `DOXYGEN_DOT_EXECUTABLE` is an un-usable state.  I.e. do we need something like:
```suggestion
      set(DOXYGEN_DOT_EXECUTABLE "${_prev_dot}")
  endif()
```
?

---

> When DOXYGEN_DOT_EXECUTABLE is not found now but was found before

This condition shouldn't be possible; once a cache variable is set, `find_executable` doesn't check or change it. I just tried by changing `DOXYGEN_DOT_EXECUTABLE` to something bogus, and the variable remained set to the same value rather than empty.

---

Then `prev_dot` is solely recorded whether it had been found before ...

---

Well, it encodes an "optional boolean": not defined, false value, or true value.

---

(I remember you prefer restricting the nesting, so this might be moot):

Would this be equivalent with 'simpler' conditional?
```
  if(_prev_dot STREQUAL "")  # Note: use string comparison, not boolean, to avoid matching old "NOTFOUND"
	  if(DOXYGEN_DOT_EXECUTABLE)
	    message(STATUS "Found dot: ${DOXYGEN_DOT_EXECUTABLE}")
	  else   
	    message(STATUS "Could NOT find graphviz generator `dot`")
	  endif()
  endif()
```

---

I think so? Stupid CMake booleans, I can't remember if you can match a copied "unset" variable with `STREQUAL ""`...